[GLIB] Make SysprofAnnotator pass null / empty strings

GeorgesStavracas opened the pull request:
#### d83571d66df9844a25c4156433a50665ab328f6d
<pre>
[GLIB] Make SysprofAnnotator pass null / empty strings
<a href="https://bugs.webkit.org/show_bug.cgi?id=277736">https://bugs.webkit.org/show_bug.cgi?id=277736</a>

Reviewed by Adrian Perez de Castro.

Instead of a fully empty string (&quot;&quot;) or no string at all (nullptr), it
uses a whitespaced string (&quot; &quot;). This makes Sysprof think there&apos;s a
description, when in fact it doesn&apos;t have.

Pass nullptr when the string is empty, and never pass whitespaced
strings as descriptions.

* Source/WTF/wtf/SystemTracing.h:
* Source/WTF/wtf/glib/SysprofAnnotator.h:

Canonical link: <a href="https://commits.webkit.org/282044@main">https://commits.webkit.org/282044@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/291ec5e71c0eb837490f73c766498d7f59f68efc

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/61781 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/41135 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/14373 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/65760 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/12326 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/48821 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/12597 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/49816 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/8552 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/64850 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/38206 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/53511 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/30648 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/34858 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/10741 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/11257 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/54876 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/56666 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/11044 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/67488 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/61022 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/5724 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/10798 "Found 1 new test failure: workers/worker-to-worker.html (failure)") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/57192 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/5749 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/53458 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/57426 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/13760 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/4707 "Passed tests") | | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/82785 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/36935 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/14483 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/38019 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/39115 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/37764 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aperezdc commented on Source/WTF/wtf/SystemTracing.h:
```diff
@@ -355,22 +355,25 @@ enum WTFOSSignpostType {
 #elif USE(SYSPROF_CAPTURE)
 
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wformat-zero-length"
```
Instead of this, let's use the `IGNORE_WARNINGS_{BEGIN,END}` macros defined by WTF:

```suggestion
IGNORE_WARNINGS_BEGIN("format-zero-length")
```

aperezdc commented on Source/WTF/wtf/SystemTracing.h:
```diff
@@ -390,6 +393,8 @@ enum WTFOSSignpostType {
 #define WTFBeginSignpostAlwaysWithTimeDelta(pointer, name, timeDelta, ...) WTFBeginSignpostWithTimeDelta((pointer), name, (timeDelta), ##__VA_ARGS__)
 #define WTFEndSignpostAlwaysWithTimeDelta(pointer, name, timeDelta, ...) WTFEndSignpostWithTimeDelta((pointer), name, (timeDelta), ##__VA_ARGS__)
 
+#pragma GCC diagnostic pop
```
Ditto.

```suggestion
IGNORE_WARNINGS_END
```

aperezdc reviewed (changes requested):
Patch LGTM but let's use the WTF macros for tweaking compiler warning settings, please check the comments.

philn commented on Source/WTF/wtf/SystemTracing.h:
```diff
@@ -355,22 +355,24 @@ enum WTFOSSignpostType {
 #define WTFEmitSignpost(pointer, name, ...) \
     do { \
         if (auto* annotator = SysprofAnnotator::singletonIfCreated()) \
-            annotator->instantMark(std::span(_STRINGIFY(name)), " " __VA_ARGS__); \
```
Move `IGNORE_WARNINGS_BEGIN("error=format-zero-length")` here?

philn commented on Source/WTF/wtf/SystemTracing.h:
```diff
@@ -355,22 +355,24 @@ enum WTFOSSignpostType {
     do { \
         if (auto* annotator = SysprofAnnotator::singletonIfCreated()) \
-            annotator->instantMark(std::span(_STRINGIFY(name)), " " __VA_ARGS__); \
+            annotator->instantMark(std::span(_STRINGIFY(name)), "" __VA_ARGS__); \
```
`IGNORE_WARNINGS_END` after this line. And ditto for each macro.

GeorgesStavracas committed:
[GLIB] Make SysprofAnnotator pass null / empty strings
https://bugs.webkit.org/show_bug.cgi?id=277736

Reviewed by Adrian Perez de Castro.

Instead of a fully empty string ("") or no string at all (nullptr), it
uses a whitespaced string (" "). This makes Sysprof think there's a
description, when in fact it doesn't have.

Pass nullptr when the string is empty, and never pass whitespaced
strings as descriptions.

* Source/WTF/wtf/SystemTracing.h:
* Source/WTF/wtf/glib/SysprofAnnotator.h:

Canonical link: https://commits.webkit.org/282044@main
